pkg/app/boot.go

```diff
@@ -1,24 +1,64 @@
 package app
 
 import (
+	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
 
 	"github.com/pkg/errors"
 )
 
 // bootedPath is the path where to store the booted flag
-var bootedPath = "/var/run/modules"
+const bootedPath = "/var/run/modules"
+
+// bootManager contains the path of the booted files and fileSystem
+type bootManager struct {
+	bootedPath string
+	fs         fileSystem
+}
+
+type fileSystem interface {
+	Create(string) (io.ReadCloser, error)
+	MkdirAll(string, uint32) error
+	Stat(string) (any, error)
+}
+
+type defaultBootManager struct{}
+
+func (fs defaultBootManager) Create(path string) (io.ReadCloser, error) {
+	return os.Create(path)
+}
+
+func (dfs defaultBootManager) MkdirAll(path string, perm uint32) error {
+	return os.MkdirAll(path, fs.FileMode(perm))
+}
+
+func (dfs defaultBootManager) Stat(path string) (any, error) {
+	return os.Stat(path)
+}
+
+func defaultMarkManager() *bootManager {
+	return &bootManager{
+		fs:         defaultBootManager{},
+		bootedPath: bootedPath,
+	}
+}
 
 // MarkBooted creates a file in a memory
 // this file then can be used to check if "something" has been restared
 // if its the first time it starts
 func MarkBooted(name string) error {
-	if err := os.MkdirAll(bootedPath, 0770); err != nil {
+	manager := defaultMarkManager()
+	return manager.markBooted(name)
```

Comment:
No need to define a manager imho. All what u need to pass is, name, path, and something that implements a `fileSystem`

```suggestion
	return markBooted(name, bootedPath, &defaultFileSystem{})
```

in fact you don't even need to do `&defaultFileSystem{}` on each call this can be a global hidden var because it's mostly what you gonna use

```go 
type defaultFileSystem struct{}

var (
   defaultFS = &defaultFileSystem{}
)

// then in your call
func MarkBooted(name string) {
   return   markBooted(name, path, defaultFS)
}
```